esphome/components/st7789v/display.py

```diff
@@ -29,55 +31,113 @@
 ST7789VRef = ST7789V.operator("ref")
 ST7789VModel = st7789v_ns.enum("ST7789VModel")
 
+MODEL_ENUM = "model_enum"
+MODEL_PRESETS = "model_presets"
+REQUIRE_PS = "require_ps"
+
+
+def model_spec(require_ps=False, presets=None):
+    if presets is None:
+        presets = {}
+    return {MODEL_PRESETS: presets, REQUIRE_PS: require_ps}
+
+
 MODELS = {
-    "TTGO_TDISPLAY_135X240": ST7789VModel.ST7789V_MODEL_TTGO_TDISPLAY_135_240,
-    "ADAFRUIT_FUNHOUSE_240X240": ST7789VModel.ST7789V_MODEL_ADAFRUIT_FUNHOUSE_240_240,
-    "ADAFRUIT_RR_280X240": ST7789VModel.ST7789V_MODEL_ADAFRUIT_RR_280_240,
-    "ADAFRUIT_S2_TFT_FEATHER_240X135": ST7789VModel.ST7789V_MODEL_ADAFRUIT_S2_TFT_FEATHER_240_135,
-    "CUSTOM": ST7789VModel.ST7789V_MODEL_CUSTOM,
+    "TTGO_TDISPLAY_135X240": model_spec(
+        presets={
+            CONF_BACKLIGHT_PIN: "GPIO4",
+            CONF_CS_PIN: "GPIO5",
+            CONF_DC_PIN: "GPIO16",
+            CONF_RESET_PIN: "GPIO23",
+            CONF_HEIGHT: 240,
+            CONF_WIDTH: 135,
+            CONF_OFFSET_HEIGHT: 52,
+            CONF_OFFSET_WIDTH: 40,
+        }
+    ),
+    "ADAFRUIT_FUNHOUSE_240X240": model_spec(
+        presets={
+            CONF_HEIGHT: 240,
+            CONF_WIDTH: 240,
+            CONF_OFFSET_HEIGHT: 0,
+            CONF_OFFSET_WIDTH: 0,
```

Comment:
We can add these here, too, I think:
```suggestion
            CONF_OFFSET_WIDTH: 0,
            CONF_CS_PIN: "GPIO40",
             CONF_DC_PIN: "GPIO39",
             CONF_RESET_PIN: "GPIO41",
```